lib/appoptics_apm/oboe_init_options.rb

```diff
@@ -99,9 +99,14 @@ def array_for_oboe
     end
 
     def service_key_ok?
-      # only required for ssl reporter
-      if @reporter == 'ssl' && @service_key !~ /^[0-9a-fA-F]{64}:[-.:_?\\\/\w ]{1,255}$/
-        AppOpticsAPM.logger.warn '[appoptics_apm/oboe_options] APPOPTICS_SERVICE_KEY problem. No service name or api token in wrong format. Cannot submit data.'
+      # only required for ssl reporter, always return true for other reporters
+      if @reporter == 'ssl' &&
+        !(@service_key =~ /^[0-9a-fA-F]{64}:[-.:_?\\\/\w ]{1,255}$/ ||   # API token
+          @service_key =~ /^[0-9a-zA-Z_\-]{71}:[-.:_?\\\/\w ]{1,255}$/ ) # SWOKEN
+
```

Comment:
I opted for a different approach here. It probably doesn't matter much at all consider it's once at startup but

1. i don't check for ssl; i just set valid keys regardless of what reporter i'm using. my thought is why add a special check just so i can avoid setting a valid key for testing. it changes the logic and optimizes and non-production flow.
2. i combined both regexes into one so i don't have two executable statements. (i probably should have put the new one first though) like: `@service_key =~ /^([A-Za-z0-9_-]{71}|[A-Fa-f0-9]{64}):[a-z0-9.:_-]{1,255}$/` but that does depend on the fact that i've already remove whitespace other invalid characters from the string via
```
const cleansedKey = serviceKey
  .replace(/ /g, '-')
  .replace(/[^A-Za-z0-9.:_-]/g, '');
```

i don't think you need to quote the trailing `-` in the SWOKEN check. leading or trailing dash in a range doesn't need quoting.

Reply:
I probably should have used `\w` like you did too!

Reply:
also, i didn't know that `\` or `/` were valid in the service name.

Reply:
I was surprised too that I had to quote that '-', but the regex would error. 

I only check the key for ssl, because the null reporter doesn't need it. It's not just for testing anymore. Generally I don't set a key for testing, unless I write to the "real" collector.